python3Packages.bugwarrior: 1.8.0 -> 2.0.0

ryneeverett opened the pull request:
Resolve #353713. Also, add me as a maintainer. Disclosure: I'm the upstream maintainer.

https://github.com/GothenburgBitFactory/bugwarrior/releases/tag/2.0.0

I added release notes since the PR template suggests it. I think this needs to be backported to 25.11 even though it's a major release since bugwarrior is otherwise broken (#353713).

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda commented on nixos/doc/manual/release-notes/rl-2511.section.md:
Move to 26.05.

dotlambda commented on pkgs/development/python-modules/bugwarrior/default.nix:
```diff
@@ -4,65 +4,64 @@
+  disabled = pythonOlder "3.10";
+  pyproject = true;
 
   src = fetchPypi {
```
fetchFromGitHub is preferred

dotlambda commented on pkgs/development/python-modules/bugwarrior/default.nix:
```diff
@@ -4,65 +4,64 @@
+    sha256 = "sha256-blTKtNfhTZyJHwIOq1P7HjtEmNnz/bL6dWr402veNm4=";
   };
 
   propagatedBuildInputs = [
```
```suggestion
  dependencies = [
```

dotlambda commented on pkgs/development/python-modules/bugwarrior/default.nix:
```diff
@@ -4,65 +4,64 @@
+    requests
+    taskw
+    tomli
+
```
```suggestion
  ]
  ++ pydantic.optional-dependencies.email
```

dotlambda commented on pkgs/development/python-modules/bugwarrior/default.nix:
```diff
@@ -4,65 +4,64 @@
   ];
 
-  # for the moment oauth2client <4.0.0 and megaplan>=1.4 are missing for running the test suite.
+  # for the moment oauth2client <4.0.0 is missing for running the test suite.
```
Can you disable the tests that require oauth2client?

dotlambda reviewed (changes requested):
Add meta.changelog and pythonImportsCheck.

dotlambda commented on nixos/doc/manual/release-notes/rl-2511.section.md:
```diff
@@ -307,6 +307,8 @@
 
 - `orjail` package has been removed as it is broken by the latest firejail release and seems unmaintained.
 
+- `bugwarrior` has been updated to major version 2.0.0 which will likely require manual intervention. See [release notes](https://github.com/GothenburgBitFactory/bugwarrior/releases/tag/2.0.0).
```
```suggestion
- `bugwarrior` was promoted to a top-level attribute and has been updated to major version 2.0.0 which will likely require manual intervention. See [release notes](https://github.com/GothenburgBitFactory/bugwarrior/releases/tag/2.0.0).
```
```suggestion
- `bugwarrior` was promoted to a top-level attribute and has been updated to major version 2.0.0 which will likely require manual intervention. See [release notes](https://github.com/GothenburgBitFactory/bugwarrior/releases/tag/2.0.0).
```

dotlambda commented on pkgs/by-name/bu/bugwarrior/package.nix:
```diff
@@ -0,0 +1,97 @@
+
+  src = fetchFromGitHub {
+    owner = "GothenburgBitFactory";
+    repo = pname;
```
```suggestion
    repo = "bugwarrior";
```
https://github.com/NixOS/nixpkgs/issues/277994

dotlambda commented on pkgs/by-name/bu/bugwarrior/package.nix:
```diff
@@ -0,0 +1,97 @@
+  src = fetchFromGitHub {
+    owner = "GothenburgBitFactory";
+    repo = pname;
+    rev = version;
```
```suggestion
    tag = version;
```

dotlambda commented on pkgs/by-name/bu/bugwarrior/package.nix:
```diff
@@ -0,0 +1,97 @@
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+  doCheck = true;
```
```suggestion
```
That's the default for Python packages.

dotlambda commented on pkgs/by-name/bu/bugwarrior/package.nix:
```diff
@@ -0,0 +1,97 @@
+      importlib-metadata
+      taskwarrior3
+    ]
+    ++ lib.lists.flatten (builtins.attrValues optional-dependencies);
```
```suggestion
    ++ lib.concatAttrValues optional-dependencies;
```

dotlambda reviewed (commented):
How about splitting this into two commits?

* bugwarrior: init at 2.0.0
* python3Packages.bugwarrior: drop

dotlambda commented on pkgs/by-name/bu/bugwarrior/package.nix:
```diff
@@ -0,0 +1,93 @@
+  };
+
+  nativeCheckInputs = [
+    writableTmpDirAsHomeHook
```
Consider also adding versionCheckHook.

dotlambda commented on pkgs/by-name/bu/bugwarrior/package.nix:
```diff
@@ -0,0 +1,93 @@
+    "TestServices"
+  ];
+
+  meta = with lib; {
```
```suggestion
  meta = {
```
https://github.com/NixOS/nixpkgs/issues/208242

dotlambda committed:
python3Packages.taskw: propagate distutils

ryneeverett committed:
python3Packages.bugwarrior: 1.8.0 -> 2.0.0

ryneeverett commented:
Thanks, that's a bug! https://github.com/GothenburgBitFactory/bugwarrior/pull/1159 I went ahead and moved importlib-metadata to dependencies.

I also reverted the move out of the python package set because I realized that plugins actually need to be able to import bugwarrior. There aren't any worth packaging at this point but bugwarrior isn't inherently a leaf package after all. Sorry for the extra review this caused.

dotlambda commented on pkgs/development/python-modules/bugwarrior/default.nix:
```diff
@@ -1,68 +1,122 @@
+    versionCheckHook
+    writableTmpDirAsHomeHook
+  ]
+  ++ [
```
```suggestion
```

dotlambda commented on pkgs/development/python-modules/bugwarrior/default.nix:
```diff
@@ -1,68 +1,122 @@
+    ruff
+    sphinx
+    sphinx-click
+    sphinx-inline-tabs
```
These are required for tests?

ryneeverett replied:
Yes, there are tests of the docs build. Those tests aren't at all helpful as a diagnostic for nixpkgs but to me it seems a little cleaner to include the dependencies than to disable them since they're already packaged.
